packages/react-button/src/components/Button/ButtonBase.tsx

```diff
@@ -10,10 +10,16 @@ export const ButtonBase = compose<'button', ButtonProps, ButtonProps, {}, {}>(
 
     return (
       <slots.root ref={ref} {...slotProps.root}>
-        {props.loading && <slots.loader {...slotProps.loader} />}
-        {props.icon && props.iconPosition !== 'after' && <slots.icon {...slotProps.icon} />}
-        {!props.iconOnly && props.content && <slots.content {...slotProps.content} />}
-        {props.icon && props.iconPosition === 'after' && <slots.icon {...slotProps.icon} />}
+        {state.children ? (
+          state.children
```

Comment:
This becomes extremely confusing 🤔 

- What I should use `state.children`, `props.children` or `slotProps.children`?
- What I should use `state.icon`, `props.icon` or `slotProps.icon`?

Reply:
I think this makes sense. I'm going to repurpose this PR to get rid of the `content` slot since that itself is confusing with `children` and we shouldn't reinvent things here and just go with what `HTML` is already doing. I'll think about how this should look.

Reply:
So, `state` should === superset of `props`, if any. That's it. The point is totally spot on though, we need to simplify and possibly remove slotProps in favor of state. It's too confusing. I don't think slotProps is necessary when state[slot] is good enough.

The only answer should be `state.children`. By default, with no state hooks, state === props.